winpr/libwinpr/sspi/Kerberos/krb5glue_mit.c

```diff
@@ -0,0 +1,186 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifdef WITH_KRB5_MIT
+
+#include <winpr/path.h>
+#include <winpr/wlog.h>
+#include <winpr/endian.h>
+#include <winpr/crypto.h>
+#include <winpr/print.h>
+#include <errno.h>
+#include "krb5glue.h"
+#include <profile.h>
+
+static char* create_temporary_file(void)
+{
+	BYTE buffer[32];
+	char* hex;
+	char* path;
+
+	winpr_RAND(buffer, sizeof(buffer));
+	hex = winpr_BinToHexString(buffer, sizeof(buffer), FALSE);
+	path = GetKnownSubPath(KNOWN_PATH_TEMP, hex);
+	free(hex);
+	return path;
+}
+
+void krb5glue_keys_free(krb5_context ctx, struct krb5glue_keyset *keyset)
+{
+    krb5_k_free_key(ctx, keyset->session_key);
+    krb5_k_free_key(ctx, keyset->initiator_key);
+    krb5_k_free_key(ctx, keyset->acceptor_key);
+}
+
+krb5_error_code krb5glue_update_keyset(krb5_context ctx, krb5_auth_context auth_ctx, BOOL acceptor, struct krb5glue_keyset *keyset)
+{
+    krb5glue_keys_free(ctx, keyset);
+    krb5_auth_con_getkey_k(ctx, auth_ctx, &keyset->session_key);
+    if (acceptor)
+    {
+        krb5_auth_con_getsendsubkey_k(ctx, auth_ctx, &keyset->acceptor_key);
+        krb5_auth_con_getrecvsubkey_k(ctx, auth_ctx, &keyset->initiator_key);
+    }
+    else
+    {
+        krb5_auth_con_getsendsubkey_k(ctx, auth_ctx, &keyset->initiator_key);
+        krb5_auth_con_getrecvsubkey_k(ctx, auth_ctx, &keyset->acceptor_key);
+    }
+    return 0;
+}
+
+krb5_prompt_type krb5glue_get_prompt_type(krb5_context ctx, krb5_prompt prompts[], int index)
+{
+    krb5_prompt_type *types = krb5_get_prompt_types(ctx);
+    return types ? types[index] : 0;
+}
+
+krb5_error_code krb5glue_log_error(krb5_context ctx, krb5_data *msg, const char *tag)
+{
+	krb5_error* error = NULL;
+    krb5_error_code rv = 0;
+
+    if (!(rv = krb5_rd_error(ctx, msg, &error)))
+    {
+        WLog_ERR(tag, "KRB_ERROR: %s", error->text.data);
+        krb5_free_error(ctx, error);
+    }
+
+    return rv;
+}
+
+BOOL krb5glue_authenticator_validate_chksum(krb5glue_authenticator authenticator, int cksumtype,  uint32_t *flags)
+{
+    if (!authenticator || !authenticator->checksum || authenticator->checksum->checksum_type != cksumtype || authenticator->checksum->length < 24)
+        return FALSE;
+    Data_Read_UINT32((authenticator->checksum->contents + 20), (*flags));
+    return TRUE;
+}
+
+krb5_error_code krb5glue_get_init_creds(krb5_context ctx, krb5_principal princ, krb5_ccache ccache, krb5_prompter_fct prompter, char *password, SEC_WINPR_KERBEROS_SETTINGS *krb_settings)
+{
+    krb5_error_code rv = 0;
+	krb5_deltat start_time = 0;
```

Comment:
strange code style here, a `clang-format` run could improve things

Reply:
Had in mind to run clang-format but forgot :( sorry